File: sdk/python/examples/http_database.py
from megacache.client import CachePolicy, MegaCacheClient
from megacache.integrations import cached_loader, wsgi_traceparent

client = MegaCacheClient()
load_user = cached_loader(
    client,
    CachePolicy(ttl_seconds=60, stale_seconds=300, tags=("users",)),
)


def application(environ, start_response):
    value = load_user("user:42", lambda: b'{"id":42}')
    start_response("200 OK", [("Content-Type", "application/json")])
    return [value.value]


application = wsgi_traceparent(client, application)

File: sdk/node/examples/http-database.js
const { MegaCacheClient } = require("../src");
const { traceparentMiddleware, cachedDatabaseLoader } = require("../src/integrations");

const client = new MegaCacheClient();
const middleware = traceparentMiddleware(client); // app.use(middleware)
const load = cachedDatabaseLoader(client, { ttlSeconds: 30, staleSeconds: 120 });
// const user = await load("user:42", () => pool.query("select ..."));
module.exports = { client, middleware, load };
